qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -231,11 +254,71 @@ def __init__(self,
             get_cmd=":FUNCtion:IMPedance:RANGe?",
             set_cmd=":FUNCtion:IMPedance:RANGe {}",
             unit='Ohm',
-            vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
+            vals=self._imp_range,
             docstring="Selects the impedance measurement range, also turns "
                       "the auto range function OFF."
         )
 
+        self.add_parameter(
+            "dc_bias_enabled",
+            get_cmd=":BIAS:STATe?",
+            set_cmd=":BIAS:STATe {}",
+            vals=Bool(),
+            val_mapping=create_on_off_val_mapping(on_val="1",
+                                                  off_val="0"),
+            docstring="Enables DC bias. DC bias is automatically turned "
+                      "off after recalling the state from memory."
+        )
+
+        self.add_parameter(
+            "dc_bias_voltage_level",
+            get_cmd=":BIAS:VOLTage:LEVel?",
+            set_cmd=":BIAS:VOLTage:LEVel {}",
+            get_parser=float,
+            unit="V",
+            vals=self._dc_bias_v_level_range,
+            docstring="Sets the DC bias voltage. Setting does not "
+                      "implicitly turn the DC bias ON."
+        )
+
+        self.add_parameter(
+            "meas_time_mode",
+            initial_value="MED",
+            vals=Enum("SHOR", "MED", "LONG"),
+            parameter_class=GroupParameter,
+            docstring="Measurement time mode holds the mode of measurement "
+                      "time. It can be SHOR (short), MED (medium - default) "
+                      "or LONG."
+        )
+
+        self.add_parameter(
+            "averaging_rate",
+            initial_value=1,
+            vals=Ints(1, 256),
+            parameter_class=GroupParameter,
+            docstring="Averaging rate for the measurement."
+        )
+
+        self._set_measurement_time = Group(
```

Comment:
a better, more specific name? also, it's just a parameter group as opposed to some cool private method that "sets measurement time", right?
```suggestion
        self._aperture_group = Group(
```

Reply:
yes